text/002-css-rules.md

```diff
@@ -0,0 +1,1094 @@
+const button = rules({
+  // base styles
+
+  variants: {
+    color: {
+      brand: { color: "#FFFFA0" },
+      accent: { color: "#FFE4B5" }
+    },
+    size: {
+      small: { padding: 12 },
+      medium: { padding: 16 },
+      large: { padding: 24 }
+    }
+  },
+  compoundVariants: [
+    {
+      color: brand,
+      size: small,
+      style: {
+        fontSize: "16px"
+      }
+    }
+  ]
+});
+```
+
+It doesn't seem uncomfortable when the conditions are not as demanding as they are now.  
+But if the conditions become complicated, it will be inconvenient to fill out.
+
+**Code After**
+
+```typescript
+const button = rules({
+  // base styles
+
+  variants: {
+    color: {
+      brand: { color: "#FFFFA0" },
+      accent: { color: "#FFE4B5" }
+    },
+    size: {
+      small: { padding: 12 },
+      medium: { padding: 16 },
+      large: { padding: 24 }
+    }
+  },
+  // method 1
+  compoundVariants: [
+    {
+      color: brand,
+      size: small,
+      style: {
+        fontSize: "16px"
+      }
+    }
+  ],
+  // method 2
+  compoundVariants: ({ color, size }) => [
+    {
+      condition: [color.brand, size.small],
+      style: {
+        fontSize: "16px"
+      }
+    }
+  ]
+});
+```
+
+The first method is the original method in stitches's compound variants, and it is still a good way for it.  
+So we keep it.
+
+And the second method we present will provide a more comfortable way to write the conditions.  
+This feature checks existing variants and provides automatic completion.
+
+**Compiled:**
+
+```css
+.[FILE_NAME]_button_compound_0__[HASH] {
+  font-size: '16px'
+}
+.[FILE_NAME]_button_compound_1__[HASH] {
+  font-size: '24px',
+  font-weight: bold
+}
+```
+
+**Usage:**
+
+If you use the already set combination, it will be applied automatically.
+
+```typescript
+button({
+  color: "accent",
+  size: "large"
+});
+```
+
+## 7. Default Variants
+
+The way of [Stitches's Default Variants](https://stitches.dev/docs/variants#default-variants) is already good to use, so we keep this method in ours.
+
+**Code:**
+
+```typescript
+const button = rules({
+  // base styles
+
+  variants: {
+    color: {
+      brand: {
+        color: "#FFFFA0",
+        backgroundColor: "blueviolet"
+      },
+      accent: {
+        color: "#FFE4B5",
+        backgroundColor: "slateblue"
+      }
+    }
+  },
+  defaultVariants: {
+    color: "brand"
+  }
+});
+```
+
+## 8. Composition
+
+**Code:**
+
+[Vanilla Extract's Composition](https://vanilla-extract.style/documentation/style-composition/) is enough to use already. So we maintain this basically.  
+And It gonna inherit base rules like [Stitches's Composing Components](https://stitches.dev/docs/composing-components).
+
+```typescript
+const baseRule = rules({
+  padding: "12px",
+
+  variants: {
+    size: {
+      small: { fontSize: "11px" },
+      large: { fontSize: "20px" }
+    }
+  }
+});
+
+const myRule = rules([baseRule, { color: "blue", marginTop: "10px" }]);
```

Comment:
> [!NOTE]
> **P2[Strongly Recommended]: Please actively consider it. Changes are strongly encouraged, and if you don't make a change, you need a good reason.**

It should be written about how the interaction will happen when there are more variants.

Reply:
Now I added more variantss with toggles option.